console/console-init/ui/src/modules/connection-detail/containers/ConnectionLinksToolbarContainer/ConnectionLinksToolbarContainer.tsx

```diff
@@ -0,0 +1,380 @@
+    setNameOptions([initalSelectOption]);
+  }
+  if (!addressOptions) {
+    setAddressOptions([initalSelectOption]);
+  }
+
+  const onFilterSelect = (value: string) => {
+    setFilterSelected(value);
+  };
+  const onNameSelect = (e: any, selection: SelectOptionObject) => {
+    setNameSelected(selection.toString());
+    setNameInput(undefined);
+  };
+  const onNameClear = () => {
+    setNameSelected(undefined);
+    setNameInput(undefined);
+  };
+  const onChangeNameInput = async (value: string) => {
+    const response = await client.query<IConnectionLinksNameSearchResponse>({
+      query: RETURN_ALL_CONNECTION_LINKS_FOR_NAME_SEARCH(
+        connectionName,
+        namespace,
+        value.trim()
+      ),
+      fetchPolicy: FetchPolicy.NETWORK_ONLY
+    });
+    if (
+      response &&
+      response.data &&
+      response.data.connections &&
+      response.data.connections.connections &&
+      response.data.connections.connections.length > 0 &&
+      response.data.connections.connections[0].links &&
+      response.data.connections.connections[0].links.links &&
+      response.data.connections.connections[0].links.links.length > 0
+    ) {
+      const obtainedList = response.data.connections.connections[0].links.links.map(
+        (link: any) => {
+          return link.metadata.name;
+        }
+      );
+      //get list of unique records to display in the select dropdown based on total records and 100 fetched objects
+      const filteredNameOptions = getSelectOptionList(
+        obtainedList,
+        response.data.connections.connections[0].links.total
+      );
+      if (filteredNameOptions.length > 0) return filteredNameOptions;
+    }
+  };
+  const onNameFilter = (e: any) => {
+    const input = e.target.value && e.target.value.trim();
+    setNameInput(input);
+    setNameOptions(undefined);
+    if (input.trim().length < TYPEAHEAD_REQUIRED_LENGTH) {
+      setNameOptions([
+        <SelectOption
+          value={TypeAheadMessage.MORE_CHAR_REQUIRED}
+          isDisabled={true}
+        />
+      ]);
+    } else {
+      onChangeNameInput(input).then(data => {
+        const list = data;
+        const options = list
+          ? list.map((object, index) => (
+              <SelectOption
+                disabled={object.isDisabled}
+                key={index}
+                value={object.value}
+              />
+            ))
+          : [];
+        if (options && options.length > 0) {
+          setNameOptions(options);
+        } else {
+          setNameOptions([
+            <SelectOption
+              value={TypeAheadMessage.NO_RESULT_FOUND}
+              isDisabled={true}
+            />
+          ]);
+        }
+      });
+    }
+    const options = [
+      <SelectOption
+        value={TypeAheadMessage.MORE_CHAR_REQUIRED}
+        key="1"
+        isDisabled={true}
+      />
+    ];
+    return options;
+  };
+  const onAddressSelect = (e: any, selection: SelectOptionObject) => {
```

Comment:
can we rename SelectOptionObject with SelectOption. try to avoid use postfix or prefix data type.

Reply:
this type is not a custom interface but it is type provided by patternfly.